pkg/storage/redis/redis.go

```diff
@@ -69,8 +69,7 @@ type Backend struct {
 }
 
 // New creates a new redis storage backend.
-func New(rawURL string, options ...Option) (*Backend, error) {
-	ctx := context.TODO()
+func New(ctx context.Context, rawURL string, options ...Option) (*Backend, error) {
```

Comment:
This is causing `ctx` to be persisted. I believe since we call this from `OnConfigChange` this will result in hanging on to changes inadvertently. We need to use a context that persists across the lifetime of the databroker.